### DISCUSSION TITLE
The future of screenFetch

### DISCUSSION THREAD

**Discussion by KittyKatt** [KittyKatt/screenFetch#805] [Announcements]
**Hello all!**

It's been a long run with this script. I first created it back in 2009 to learn BASH scripting as I graduated from high school and I'm 33 years old now. Over those years it sorta evolved into the...mess it is now. As 2019 came and went I found myself having less and less time to devote to this until at some point the issues and PRs had piled up so much that just trying to sift through them was overwhelming. I had some help along the way (huge shoutout to @darealshinji for all the hard work they put in for...multiple years?), but it still wasn't quite enough to keep the boat above water.

In 2021, I decided that I really wanted to pick the work back up and start contributing to this again. I'd learned things and I wanted to use that knowledge! However, I found that working on screenFetch as it was had become cumbersome and difficult as it had evolved into this massive behemoth of BASH that I'd put together over the course of 10 years. I'd learned a lot of things over that time and as I grew in my knowledge, the way I wrote new features evolved as well. Looking over the whole 6800 (!!!) lines of BASH, I started to realize something: it was looking beyond salvation to me. So I went about trying to recreate screenFetch with new ideas and a new framework in mind. Thus, [fetch](https://github.com/KittyKatt/fetch) was born.

This went really well for a while! I made a ton of progress, created workflows to automate deb/rpm packaging, wrote some tests, all kinds of new stuff! I soon found, though, that even this had become too much to fit into my life. The last 5 years for me have been hectic and difficult to put it *extremely* lightly. The development on fetch slowly ground to a halt after just 2 months of sustained work on it. It was soon also abandoned.

I'm still on GitHub *quite* frequently (probably at least once daily) and seeing the issues and PRs continue to come in (and people continuing to star the repository!) has filled me with an increasing amount of sadness. I remember one issue from over a year ago now from someone just asking me to do another release. A simple task, right? I still couldn't bring myself to open VSCode and actually put in the 5 minutes it took to do so.

Today, I created the `v3.9.9` release to help package maintainers get up to date with latest main branch. It has taken me a long time to admit it to myself, but I think the stage at which screenFetch is now is the stage it will continue to be in the future. With a heavy heart, I'd like to say that this will likely be the last release of screenFetch unless I suddenly am gripped by motivation to work on it or `fetch` again.

Not all is completely dead, though!  There are a lot of issues and a fair few PRs outstanding. I'd like to take the time that I can find energy for it and slowly work through these. My goal is to some day soon have both of those numbers at zero. It is then that I will likely archive the repository if I'm still feeling as if my life doesn't have room for this.

It's been a fun ride and I can't believe over those years this has amassed 3900 stars (a monumentally larger number than I *ever* thought it would), 495 forks, and a large handful of watchers. This was made by a high schooler just starting to learn about *nix like operating systems (and really computers in general). I never thought back then that it would ever evolve into something like this and become actually popular, let alone be used by anyone besides myself. It's really come a long way since being a small BASH script on an Arch Linux forums posts back in 2009. And I have the support of every person that opened an issue, submitted a PR, starred the repo, or even reached out to me about it to thank for the motivation I did manage to drum up from time to time.

Humbly, thank you for all the fish,
KittyKatt

**Comment by geerlingguy** [comment]
It's only just now I realized you're also from STL! Thanks for all your work on this handy utility. I just love having a way to get a quick system description I can copy and paste elsewhere, and `screenfetch` has filled that void after `neofetch`'s demise!

If you want to set up a GitHub Sponsors, that could also maybe help, though to be realistic it's a lot of work to get only a few regular sponsors :(

I'd still like to throw over a few dollars your way in appreciation for the tool with zero obligations and zero expectation for any future development!

**Reply by KittyKatt** [reply]
Wow! Honestly kinda floored you're here. Yeah, I've been following you as another St. Louis native for a pretty long while now and I love your content. :heart: I'm so happy to hear you use it at all. I remember when neofetch came out (and dylanaraps posting a very accurate criticism on screenfetch itself) and I noticed a sharp decline in screenfetch usage on Reddit and such. Mostly because neofetch (and their other later project pfetch) were genuinely really good and well thought out...everything that screenFetch wasn't lol. By that time, though, I was already starting to slow down so it wasn't too much of a disappointment.

I know my handle has been recognized a very small handful of times by randoms that I've met at a conference or some other tech event. It's so nice to hear from actual users, especially ones I look up to admire! Thank you very much for commenting.

I'll think about the Sponsor idea. It might be worthwhile and could actually motivate me away from my homelab or other various projects.

UPDATE: Well, I went ahead and made it anyways. So I have a sponsor page now: https://github.com/sponsors/KittyKatt

**Reply by C0rn3j** [reply]
> screenfetch has filled that void after neofetch's demise!

My pipeline was screenfetch -> neofetch -> fastfetch.

I would suggest to check [fastfetch](https://github.com/fastfetch-cli/fastfetch) out, it's better in every regard I could find compared to both `screenfetch` and `neofetch`, and it has profiles for both of them if you wish to keep the look you're used to.

i.e. `fastfetch -c screenfetch`

The only negative one could see is that it is no longer to as easy to contribute to, as it is written in Rust, but that's also the reason why it takes miliseconds as opposed to seconds to render.

**Comment by WinGitX86** [comment]
Hey man,

I just read your latest message about screenFetch. Made me a bit sad, but I totally get it. Life gets busy, and maintaining a 6800-line Bash script for over a decade is already insane dedication.

I'm just a random user from China, and I've been using screenFetch for years. It's small, it's fun, and it works. Thank you for all those late nights you put into this.

Don't feel guilty about the issues or PRs. You've already done more than enough. Take care of yourself first.

Wish: everything goes well.

**Comment by elacerda** [comment]
Hi there... With AI help I've created the [AstroFetch](https://github.com/elacerda/astrofetch) inspired by your screenfetch, thanks! It spawns random generated ASCII/ANSI-colored galaxies :D Thank you for your endeavour! I've always used screenfetch!
